nexus-sdk/src/main/java/io/nexusrpc/handler/OperationStartResult.java

```diff
@@ -118,9 +121,6 @@ public OperationStartResult<R> build() {
       if (syncResult != null && asyncOperationId != null) {
         throw new IllegalStateException("Cannot have both sync result and async operation ID");
       }
-      if (syncResult == null && asyncOperationId == null) {
```

Comment:
Hrmm, this case can still occur. I am not sure we should treat the default `OperationStartResult` builder as a null sync result. Maybe we should have some boolean that tracks whether it is sync (i.e. that `setSyncResult(null)` was even called)? Don't think we'd want `Optional<R>` because `Optional` is usually best for return types. You'd have to put this boolean field in the `OperationStartResult` class too since builders can be created from results. Thoughts?

Reply:
Other option is we get rid of `newBuilder`, `setAsyncOperationId`, `setSyncResult` and replace with `newSyncBuilder` and `newAsyncBuilder` so we know the result is either sync or async

Reply:
I'd be ok with that. Anything that makes the user choose between sync and async I think works.

Reply:
OK added dedicated sync and async methods to create the builder and made the builders methods private